### PR TITLE
Fix failing sql postsubmit tests

### PR DESCRIPTION
**Pull request by jasonvigil**
Always return boolean fields in export (even if value is false). Also, specify regional availability type in locationPreference test cases so that the secondaryZone is utilized.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
   databaseVersion: POSTGRES_16
   region: us-central1
   settings:
+    availabilityType: REGIONAL
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml:25]: Is this an immutable field?

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve
/hold

One question otherwise look good. Feel free to unhold

File: pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml (hunk 1 above)
Comment (jasonvigil) [reply pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml:25]: It is actually mutable. I just updated the test case to cover mutating this field also.

**Comment by jasonvigil** [comment]
/hold cancel

@yuwenma can you please re-review? Rebased and updated the test to cover mutating the `availabilityType` field.

File: pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml (hunk 1 above)
Comment (yuwenma) [reply pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml:25]: Good to know! Normally location has more constraints. Have you run against real GCP?

**Comment by jasonvigil** [comment]
Thanks @yuwenma, yes, I did run against real GCP. The differences between mock / real are minimal

File: pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml (hunk 1 above)
Comment (jasonvigil) [reply pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-locationpreference/update.yaml:25]: Yes.

**Pull request merged**